spec/queries/dunning_campaigns_query_spec.rb

```diff
@@ -0,0 +1,86 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe DunningCampaignsQuery, type: :query do
+  subject(:result) do
+    described_class.call(organization:, pagination:, search_term:, filters:, order:)
+  end
+
+  let(:pagination) { nil }
+  let(:search_term) { nil }
+  let(:filters) { nil }
+  let(:order) { nil }
+  let(:membership) { create(:membership) }
+  let(:organization) { membership.organization }
+  let(:dunning_campaign_first) do
+    create(:dunning_campaign, organization:, name: "defgh", code: "11", applied_to_organization: true)
+  end
+  let(:dunning_campaign_second) do
+    create(:dunning_campaign, organization:, name: "abcde", code: "22", applied_to_organization: true)
```

Comment:
I have seen there are two campaigns set as applied to organization... this should not happen, afaik, we could be one and only one campaign applied to the organization.

We should add a unique index on dunning_campaigns table with organization_id and applied_to_organzation: true to avoid this to happen. What do you think?

Reply:
Well, yep it's not possible right now - I haven't created the index because I have the feeling that we could apply multiple campaigns in the future.

But maybe the pragmatic approach is to add the unique index now and remove it when/if needed in the future.

Reply:
Yep - Let's do this 👍